go-controller/pkg/cni/cni.go

```diff
@@ -65,22 +64,20 @@ func podDescription(pr *PodRequest) string {
 	return fmt.Sprintf("[%s/%s]", pr.PodNamespace, pr.PodName)
 }
 
-func (pr *PodRequest) cmdAdd(kclient kubernetes.Interface) ([]byte, error) {
+func (pr *PodRequest) cmdAdd(podLister corev1listers.PodLister) ([]byte, error) {
 	namespace := pr.PodNamespace
 	podName := pr.PodName
 	if namespace == "" || podName == "" {
 		return nil, fmt.Errorf("required CNI variable missing")
 	}
 
-	kubecli := &kube.Kube{KClient: kclient}
-
 	// Get the IP address and MAC address from the API server.
 	// Exponential back off ~32 seconds + 7* t(api call)
 	var annotationBackoff = wait.Backoff{Duration: 1 * time.Second, Steps: 7, Factor: 1.5, Jitter: 0.1}
 	var annotations map[string]string
 	var err error
 	if err = wait.ExponentialBackoff(annotationBackoff, func() (bool, error) {
```

Comment:
Do we need this backoff now that the info is coming from the cache instead of the API server? I would imagine this was there to account for a loaded API server.

Reply:
we have to wait until the pod is in the cache, the difference now is that the cache content is pushed by the apiserver because we have a watcher, we were polling until now

Reply:
ah, ok. That makes sense. Thanks for explaining it to me!